sys/sched.c

```diff
@@ -204,20 +204,24 @@ void preempt_disable(void) {
   td->td_pdnest++;
 }
 
+void sched_maybe_switch(void) {
+  thread_t *td = thread_self();
+  if (!preempt_disabled())
```

Comment:
```c
if (preempt_disabled())
  return;
```
General flow in well-coded C routine is (1) filter out edge cases (2) solve the problem.

Reply:
Done.